src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -24,6 +28,7 @@ pub(crate) struct BinaryContainerWriter_1_1<'value, 'top> {
     allocator: &'top BumpAllocator,
     encoder: ContainerEncodingKind<'value, 'top>,
     value_writer_config: ValueWriterConfig,
+    macros: &'value MacroTable,
```

Comment:
🪧 There are lots of new function parameters simply plumbing through a reference to the macro table so nested value writers can invoke macros as needed.